airflow/api/common/experimental/mark_tasks.py

```diff
@@ -102,6 +102,10 @@ def set_state(
     if dag is None:
         raise ValueError("Received tasks with no DAG")
 
+    latest_execution_date = dag.latest_execution_date
+    if latest_execution_date is None:
+        return []
+
```

Comment:
This whole snippet should be modified inside `get_execution_dates`. 
https://github.com/apache/airflow/blob/2eff0e750f5a0cfde6bb68af215631f781e8fd90/airflow/api/common/experimental/mark_tasks.py#L248-L250

Also, isn't the original behavior (throwing an error) intended?

Reply:
user is doing simple click from UI, nothing selected so why should they see nasty error?

Reply:
Double-checked this. I agree that it should not throw an (internal?) error. I am wondering what's the reason for this error handling @jmcarp @ashb @kaxil @potiuk as this is only used by the frontend `airflow/www/views.py`.

Reply:
That error message is a bit of a misnomer. It used to look like this:

```diff
-    assert latest_execution_date is not None
+    if latest_execution_date is None:
+        raise ValueError("Received non-localized date {}".format(execution_date))
```

I agree with @xinbinhuang - this change should go in `get_execution_dates` (and check the value of `dates` after the call to short circuit the rest of the function.